sql/src/main/java/io/crate/exceptions/MultiException.java

```diff
@@ -61,7 +66,31 @@ public void printStackTrace(PrintWriter s) {
     @Override
     public String getMessage() {
         StringJoiner joiner = new StringJoiner("\n");
-        exceptions.forEach(e -> joiner.add(e.getMessage()));
+        for (Throwable e : exceptions) {
+            joiner.add(e.getMessage());
+            // Reduce the amount of characters returned to consume less memory
+            // this could escalate on a huge amount of exceptions that are raised by affected shards in the cluster
+            if (joiner.length() > MAX_EXCEPTION_LENGTH) {
+                break;
+            }
+        }
         return joiner.toString();
     }
+
+    public Collection<? extends Throwable> getExceptions() {
```

Comment:
only used in a test, either test it differently or at least use it inside `of(...)` and mark it visible for testing